File: Codes in C++/Array/Merge Two Sorted Array in to first array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int p=m-1;
        int q=n-1;
        for(int i=m+n-1;i>=0;i--)
        {   
            if(p<0)
                nums1[i]=nums2[q--];
            else if(q<0)
                nums1[i]=nums1[p--];
            else if(nums1[p]<nums2[q])
                nums1[i]=nums2[q--];
            else if(nums1[p]>=nums2[q])
                nums1[i]=nums1[p--];
            cout<<"p"<<p<<"q"<<q<<"arr"<<nums1[i]<<endl;
        }
        return;
    }
};

File: Codes in Python3/Array/Merge Two Sorted Array in to first array.py
class Solution:
    def merge(self, A: List[int], m: int, B: List[int], n: int) -> None:
        a, b, write_index = m-1, n-1, m + n - 1

        while b >= 0:
            if a >= 0 and A[a] > B[b]:
                A[write_index] = A[a]
                a -= 1
            else:
                A[write_index] = B[b]
                b -= 1

            write_index -= 1